### PR TITLE
Update tessellation to 2.4.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by marcinwadon**

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1 @@
+use flake
```
Comment (TheMMaciek) [inline .envrc:1]: Should that be a part of the PR?

File: src/main/resources/application.conf [hunk 2]
```diff
@@ -1,6 +1,7 @@
 snapshotStreaming {
   lastSnapshotPath = "lastSnapshot.json"
   lastIncrementalSnapshotPath = "lastIncrementalSnapshot.json"
+  environment = Testnet
```
Comment (TheMMaciek) [inline src/main/resources/application.conf:4]: that's the default config?

File: .envrc (hunk 1 above)
Comment (marcinwadon) [reply .envrc:1]: yep :+1:

File: src/main/scala/org/constellation/snapshotstreaming/FileBasedLastIncrementalGlobalSnapshotStorage.scala [hunk 3]
```diff
@@ -29,15 +29,17 @@ import io.circe.syntax._
...
+  def make[F[_]: Async: Files: KryoSerializer: Hasher](
+    path: Path,
+    hashSelect: HashSelect
   ): F[LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo]] =
     Semaphore[F](1)
-      .map(make(path, _))
+      .map(make(path, _, hashSelect))
 
-  private def make[F[_]: Async: Files: KryoSerializer](
+  private def make[F[_]: Async: Files: KryoSerializer: Hasher](
```
Comment (TheMMaciek) [inline src/main/scala/org/constellation/snapshotstreaming/FileBasedLastIncrementalGlobalSnapshotStorage.scala:39]: Is `KryoSerializer` still needed here given you added Hasher?

File: src/main/resources/application.conf (hunk 2 above)
Comment (marcinwadon) [reply src/main/resources/application.conf:4]: that's the example config, as far as I can see:
```
  node {
    l0Peers = ["""{"id": "<id>", "ip": "<ip>", "port": "<port>"}"""]
    pullInterval = "5s"
    pullLimit = 4
    # terminalSnapshotOrdinal = 30
  }
  opensearch {
    host = "http://sample"
    port = "80"
```

File: src/main/scala/org/constellation/snapshotstreaming/FileBasedLastIncrementalGlobalSnapshotStorage.scala (hunk 3 above)
Comment (marcinwadon) [reply src/main/scala/org/constellation/snapshotstreaming/FileBasedLastIncrementalGlobalSnapshotStorage.scala:39]: No idea. Probably I'd have to change the build parameters as you did in tessellation, to see that. Right?

Comment (TheMMaciek) [reply src/main/scala/org/constellation/snapshotstreaming/FileBasedLastIncrementalGlobalSnapshotStorage.scala:39]: Or just try removing it and see if it compiles.

**Comment by TheMMaciek** [comment]
LGTM

**Review by TheMMaciek — APPROVED** [review]

File: src/main/scala/org/constellation/snapshotstreaming/FileBasedLastIncrementalGlobalSnapshotStorage.scala (hunk 3 above)
Comment (marcinwadon) [reply src/main/scala/org/constellation/snapshotstreaming/FileBasedLastIncrementalGlobalSnapshotStorage.scala:39]: Ok, I'll try that in a separate PR

**Pull request merged**